File: facade/src/main/java/org/tdf/sunflower/types/BlockCreateResult.java
package org.tdf.sunflower.types;

import lombok.Value;

import java.util.Collections;
import java.util.List;

@Value
public class BlockCreateResult {
    public static final BlockCreateResult EMPTY = new BlockCreateResult(null, Collections.emptyList());
    Block block;
    List<TransactionInfo> infos;

    public static BlockCreateResult empty() {
        return EMPTY;
    }

    public boolean isEmpty() {
        return this == EMPTY;
    }
}

File: facade/src/main/java/org/tdf/sunflower/types/BlockCreateResult.kt
package org.tdf.sunflower.types

import org.tdf.sunflower.facade.TransactionInfo

data class BlockCreateResult(val block: Block?, val indices: List<TransactionInfo>) {
    val isEmpty: Boolean
        get() = this === EMPTY || block == null

    companion object {
        val EMPTY = BlockCreateResult(null, emptyList())

        @JvmStatic
        fun empty(): BlockCreateResult {
            return EMPTY
        }
    }
}